rfcs/text/0016_buildkite.md

```diff
@@ -0,0 +1,925 @@
+
+Pros:
+
+- Overall pretty powerful, pipelines execute Groovy code at runtime, so pipelines can do a lot and can be pretty complex, if you're willing to write the code
+- Pipeline changes can be tested in PRs
+- Shared Libraries allow shared code to be used across pipelines easily
+
+Cons:
+
+- The sandbox is pretty difficult to work with. There's a [hard-coded list](https://github.com/jenkinsci/script-security-plugin/tree/e99ba9cffb0502868b05d19ef5cd205ca7e0e5bd/src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists) of allowed methods for pipelines. Other methods must be approved separately, or put in a separate shared repository that runs trusted code.
+- Pipeline code is serialized by Jenkins, and the serialization process leads to a lot of issues that are difficult to debug and reason about. See [JENKINS-44924](https://issues.jenkins.io/browse/JENKINS-44924) - `List.sort()` doesn't work and silently returns `-1` instead of a list
+- Reasonably complex pipelines are difficult to view in the UI ([see above](#surfaces-information-intuitively-2))
+- Using Pipelines to manage certain configurations (such as Build Parameters) requires running an outdated job once and letting it fail to update it
+- Jobs that reference a pipeline have to be managed separately. Only third-party tools exist for managing these jobs as code (JJB and Job DSL).
+- Very difficult to test code without running it live in Jenkins
+
+#### Advanced Pipeline logic
+
+See above section. Jenkins supports very advanced pipeline logic using scripted pipelines and Groovy.
+
+#### Cloud-friendly pricing model
+
+Given that Jenkins is open-source, we pay only for infrastructure and people to manage it.
+
+#### Public access
+
+- Fine-grained authorization settings
+- Anonymous user access
+- Per-job authorization, so some jobs can be private
+
+#### Secrets handling
+
+- Supports [Credentials](https://www.jenkins.io/doc/book/using/using-credentials/), which are stored encrypted on disk and have authorization settings
+  - Credentials are difficult to manage in an automated way
+- Pipeline support for accessing credentials
+- Credentials masked in log output
+- Support for masking custom values in log output
+
+#### Support or Documentation
+
+Documentation for Jenkins is notoriously fragmented. All major functionality is provided in plugins, and documentation is spread out across the Jenkins Handbook, the CloudBees website, JIRA issues, wikis, GitHub repos, JavaDoc pages. Many plugins have poor documentation, and source code often has to be read to understand how to configure something.
+
+CloudBees offers paid support, but we're not familiar with it at this time.
+
+#### Scheduled Builds
+
+Jenkins supports scheduled builds via a Cron-like syntax, and can spread scheduled jobs out. For example, if many jobs are scheduled to run every day at midnight, a syntax is available that will automatically spread the triggered jobs evenly out across the midnight hour.
```

Comment:
Might be worth mentioning these need defined in a repository outside Kibana.

Reply:
We actually could move them into the repo if we used JJBB. I never took the time to migrate our job definitions since we were going to be moving to TeamCity.